hail/src/main/scala/is/hail/expr/ir/functions/UtilFunctions.scala

```diff
@@ -18,17 +18,33 @@ object UtilFunctions extends RegistryFunctions {
 
   def parseInt64(s: String): Long = s.toLong
 
-  def parseFloat32(s: String): Float = s match {
-    case "nan" => Float.NaN
-    case "inf" => Float.PositiveInfinity
-    case "-inf" => Float.NegativeInfinity
+  private val NAN = 1
+  private val POS_INF = 2
+  private val NEG_INF = 3
+
+  def parseSpecialNum(s: String): Int = s.length match {
```

Comment:
This will probably generate horrible bytecode. I doubt it's a bottleneck right now, so fine at the moment.